src/Handler.cpp

```diff
@@ -0,0 +1,77 @@
+#include "Handler.h"
+
+#include <roboteam_utils/Timer.h>
+#include <sstream>
+#include <roboteam_proto/messages_robocup_ssl_wrapper.pb.h>
+
+void Handler::start() {
+    init();
+    setupSSLClients();
+
+    roboteam_utils::Timer t;
+
+    t.loop(
+        [&]() {
+            auto vision_packets = receiveVisionPackets();
+            auto referee_packets = receiveRefereePackets();
+            std::vector<proto::RobotData> robothub_info;
+            {
+                std::lock_guard guard(sub_mutex);
+                robothub_info = receivedRobotData;
+                receivedRobotData.clear();
+            }
+
+            auto state = observer.process(Time::now(),vision_packets,referee_packets,robothub_info); //TODO: fix time extrapolation
+            while(!pub_state->send(state)); //TODO: try sending for a max limit amount of times
+        },
+        100);
+}
+
+void Handler::init() {
+    pub_state = std::make_unique<proto::Publisher<proto::State>>(proto::WORLD_CHANNEL);
+    //TODO: update channel type
+    sub_feedback = std::make_unique<proto::Subscriber<proto::RobotData>>(proto::FEEDBACK_PRIMARY_CHANNEL,&Handler::robotDataCallBack,this);
+    sub_feedback_2 = std::make_unique<proto::Subscriber<proto::RobotData>>(proto::FEEDBACK_SECONDARY_CHANNEL,&Handler::robotDataCallBack,this);
+
+}
+
+void Handler::setupSSLClients() {
+    constexpr int DEFAULT_VISION_PORT = 10006;
+    constexpr int DEFAULT_REFEREE_PORT = 10003;
+
+    const string SSL_VISION_SOURCE_IP = "224.5.23.2";
+    const string SSL_REFEREE_SOURCE_IP = "224.5.23.1";
+    
```

Comment:
Put these in constants?

Reply:
? WDYM. They are constants already?

Reply:
I think he meant Constants.cpp

Reply:
Yes that, and could mark them constexpr

Reply:
There's no Constants.cpp in Roboteam_world. It's only a target of Roboteam_AI

Reply:
marking them constexpr should be fine though.